week2/starter/ColorPicker/Views/ContentView.swift

```diff
@@ -0,0 +1,111 @@
+/// frameworks are governed by their own individual licenses.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import SwiftUI
+
+struct ContentView: View {
+  @State private var alertIsVisible: Bool = false
+  @State private var redColor: Double = Constants.Genral.initColorVal
+  @State private var greenColor: Double = Constants.Genral.initColorVal
+  @State private var blueColor: Double = Constants.Genral.initColorVal
+  @State private var foregroundColor = Color(red: Constants.Genral.initColorVal, green: Constants.Genral.initColorVal, blue: Constants.Genral.initColorVal)
+  @State private var isEditing = false
+
+  var body: some View {
+
+    VStack {
+      Text("Color Picker")
+        .font(.largeTitle)
+        .foregroundColor(Color("TextColor"))
+
+      RoundedRectangle(cornerRadius: Constants.Genral.colorPickerCornerRadius)
+        .foregroundColor(foregroundColor)
+        .border(.black)
+      SliderView(lebelText: "Red", accentCol: Color.red, sliderColor: $redColor)
+      SliderView(lebelText: "Green", accentCol: Color.green, sliderColor: $greenColor)
+      SliderView(lebelText: "Blue", accentCol: Color.blue,sliderColor: $blueColor)
+      SetColorButton(redColor: redColor, greenColor: greenColor, blueColor: blueColor, foregroundColor: $foregroundColor)
+    }
+    .background(Color("BackgroundColor"))
+    .padding(Constants.Genral.padding)
+
+  }
+
+}
+
+struct SliderView: View {
+  var lebelText: String
+  var accentCol: Color
+  @Binding var sliderColor: Double
+
+  var body: some View {
+    VStack {
+      SliderLabelTextView(text: lebelText)
+      HStack {
+        Slider(value: $sliderColor, in: Constants.Genral.initColorVal...Constants.Genral.maxColorVal)
+          .accentColor(accentCol)
+        SliderLabelTextView(text: "\(Int(sliderColor.rounded()))")
+      }
+    }
+  }
+}
+
+struct SetColorButton: View {
+  var redColor: Double
+  var greenColor: Double
+  var blueColor: Double
+  @Binding var foregroundColor: Color
+
+  var body: some View {
+    Button("Set Color") {
+      foregroundColor = Color(red: redColor / Constants.Genral.maxColorVal, green: greenColor / Constants.Genral.maxColorVal, blue: blueColor / Constants.Genral.maxColorVal)
+    }
+    .padding(Constants.Genral.padding)
+    .background(
+      ZStack {
+        Color("ButtonColor")
+        LinearGradient(gradient: Gradient(colors: [Color.white.opacity(Constants.Genral.buttonColorOpacity), Color.clear]), startPoint: .top, endPoint: .bottom)
```

Comment:
There's a new `.gradient` property in Color that you can use to use the default colors with a default gradient.